app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -152,12 +152,15 @@ class TransactionStateManager extends EventEmitter {
     // or rejected tx's.
     // not tx's that are pending or unapproved
     if (txCount > txHistoryLimit - 1) {
-      const index = transactions.findIndex((metaTx) => {
-        return getFinalStates().includes(metaTx.status)
-      })
-      if (index !== -1) {
-        transactions.splice(index, 1)
-      }
+      do {
+        const index = transactions.findIndex((metaTx) => {
+          return getFinalStates().includes(metaTx.status)
```

Comment:
Similarly here, should we cache the final states array instead of creating a new one per tx?

Reply:
This is fine for now